proposals/0072-feature-gate-threshold-automation.md

```diff
@@ -0,0 +1,247 @@
+found in this feature tracking issue:
+<https://github.com/solana-labs/solana/issues/33547>.
+
+### Overview
+
+Consider two arbitrary versions of the Solana runtime, where the newer version
+has more features in its feature set.
+
+| v1.14.25  | v1.16.0   |
+| --------- | --------- |
+| Feature A | Feature A |
+|           | Feature B |
+| Feature C | Feature C |
+|           | Feature D |
+
+The runtime should activate each of these features when a strong majority of
+the network supports it.
+"Strong majority" in this context is defined by the current feature activation
+process as 95% of stake.
+“Support” for a feature means that a node is running a version of the Solana
+runtime where the code required for the feature is present. For instance, on a
+cluster with 50% of stake running v1.14.25 and 50% of stake running v1.16.0,
+only Features A and C should be eligible for activation.
+
+This SIMD proposes 3 steps over the span of one epoch to determine which
+feature gates are queued for activation and assess stake support:
+
+1. On an epoch boundary, the runtime generates a list of feature-gates queued
+for activation.
+2. Validators signal which of the queued feature-gates they support.
+3. On the next epoch boundary, the runtime activates the feature-gates that
+have the necessary stake support
+
+### Representing the Current Feature Set
+
+As mentioned above, a BPF program is being deployed that will replace the
+non-existent account `Feature111111111111111111111111111111111111`, which is
+the owner of all feature accounts.
+
+The Feature Gate program will utilize two PDAs to track the authoritative
+feature set queued for activation. One PDA (the Feature-Request PDA) will
+store newly requested feature activations and the other (the Feature-Queue
+PDA) will store the previous epoch’s requested activations.
+
+When a new epoch begins, the Feature-Request PDA will be an empty list. When
+key-holders submit the CLI command to activate a feature, the Feature Gate
+Program appends the feature ID to the Feature-Request set at the same time as
+creating the feature account . Revoked features are removed from the
+Feature-Request list. At the end of the epoch, the Feature-Request set is
+written to the feature queue PDA by the runtime, where it becomes immutable.
+The Feature-Request PDA is then reset to an empty list again, also by the
+runtime.
+
+The newly created immutable Feature-Queue can then be used by nodes to signal
+support for potential activation in the next epoch.
+
+Note that for any given epoch, the epoch will begin with an immutable set of
+previously requested feature activations, and end with the activation of those
+with enough support. This process takes one epoch. Therefore, it will take at
+least one more epoch than it currently does from the time a key-holder submits
+a feature for activation via CLI to when it is actually activated.
+
+### Signaling Support for Features
+
```

Comment:
Suppose I downgrade my node, mid-epoch for some reason (say, performance degradation). Can you downgrade your support?

Reply:
You'd be restarting after downgrading so you'd send a new signal transaction, with your updated bitmask!

We'd just want to make sure it always overwrites, which I can mention in the proposal.